config-model/src/test/java/com/yahoo/searchdefinition/processing/ImportedFieldsTestCase.java

```diff
@@ -235,6 +235,95 @@ private static Search buildChildSearch(String parentSdContent, String sdContent)
         return builder.getSearch("child");
     }
 
+    private static class ParentPosSdBuilder {
+        public String build() {
+            return joinLines("search parent {",
+                    "  document parent {",
+                    "field pos type position {",
+                    "indexing: attribute | summary",
+                    "    }",
+                    "  }",
+                    "}");
+        }
+    }
+
+    private static class ChildPosSdBuilder {
+        private boolean import_pos;
+        private boolean import_pos_zcurve_before;
+        private boolean import_pos_zcurve_after;
+
+        public ChildPosSdBuilder() {
+            import_pos = true;
+            import_pos_zcurve_before = false;
+            import_pos_zcurve_after = false;
+        }
+
+        ChildPosSdBuilder import_pos(boolean v) { import_pos = v; return this; }
+        ChildPosSdBuilder import_pos_zcurve_before(boolean v) { import_pos_zcurve_before = v; return this; }
+        ChildPosSdBuilder import_pos_zcurve_after(boolean v) { import_pos_zcurve_after = v; return this; }
+
+        public String build() {
+            return joinLines("search child {",
+                    "  document child {",
+                    "    field parent_ref type reference<parent> {",
+                    "      indexing: attribute | summary",
+                    "    }",
+                    "  }",
+                    importPosZCurve(import_pos_zcurve_before),
+                    importPos(import_pos),
+                    importPosZCurve(import_pos_zcurve_after),
+                    "}");
+        }
+
+        private static String importPos(boolean doImport) {
+            return doImport ? "import field parent_ref.pos as my_pos {}" : "";
+        }
+
+        private static String importPosZCurve(boolean doImport) {
+            return doImport ? "import field parent_ref.pos_zcurve as my_pos_zcurve {}" : "";
+        }
+    }
+
+    private static void checkPosImport(ParentPosSdBuilder parentBuilder, ChildPosSdBuilder childBuilder) throws ParseException {
+        Search search = buildChildSearch(parentBuilder.build(), childBuilder.build());
+        assertEquals(1, search.importedFields().get().fields().size());
+        assertSearchContainsImportedField("my_pos_zcurve", "parent_ref", "parent", "pos_zcurve", search);
+    }
+
+    @Test
+    public void check_pos_import() throws ParseException {
+        checkPosImport(new ParentPosSdBuilder(), new ChildPosSdBuilder());
+    }
+
+    private void expectPosImportFailure() {
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage("For search 'child', import field 'my_pos_zcurve': Field 'pos_zcurve' via reference field 'parent_ref': Field already imported");
+    }
+
+    @Test
+    public void check_pos_import_after_pos_zcurve_import() throws ParseException {
```

Comment:
Consider simplify testing. I think the only fail case we need is where we try to import the zcurve field in accordance with the position field (which should fail).